tests/marray/marray_operators.h

```diff
@@ -0,0 +1,491 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#ifndef SYCLCTS_TESTS_MARRAY_MARRAY_OPERATOR_H
+#define SYCLCTS_TESTS_MARRAY_MARRAY_OPERATOR_H
+
+#include "../common/common.h"
+#include "../common/disabled_for_test_case.h"
+#include "../common/section_name_builder.h"
+#include "marray_common.h"
+#include "marray_operator_helper.h"
+
+#include <valarray>
+
+namespace marray_operators {
+
+template <typename DataT, typename NumElementsT>
+struct operators_helper {
+  static constexpr std::size_t NumElements = NumElementsT::value;
+  using marray_t = sycl::marray<DataT, NumElements>;
+  using varray_t = std::valarray<DataT>;
+
+  template <typename init_func, typename array_type>
+  static void init(array_type &ma) {
+    for (std::size_t i = 0; i < NumElements; i++) {
+      ma[i] = init_func::template init<DataT>(i);
+    }
+  }
+
+  template <typename init_func>
+  static void init(DataT &d) {
+    d = init_func::template init<DataT>();
+  }
+};
+
+/**
+ * @brief Define several sequences to initialize array instances. */
+
+struct seq_inc {
+  template <typename T>
+  static T init(std::size_t i) {
+    return T(i + 1);
+  }
+};
+
+template <std::size_t NumElements>
+struct seq_dec {
+  template <typename T>
+  static T init(std::size_t i) {
+    return T(NumElements - i);
+  }
+};
+
+struct seq_one {
+  template <typename T>
+  static T init(std::size_t) {
+    return T(1);
+  }
+};
+
+struct seq_two {
+  template <typename T>
+  static T init(std::size_t) {
+    return T(2);
+  }
+};
+
+template <std::size_t NumElements>
+inline auto get_sequences() {
+  return named_type_pack<seq_inc, seq_dec<NumElements>, seq_one,
+                         seq_two>::generate("incrementing sequence",
+                                            "decrementing sequence",
+                                            "sequences of ones",
+                                            "sequence of twos");
+}
+
+/**
+ * @brief Define several constants to initialize scalar instances. */
+
+struct sca_one {
+  template <typename T>
+  static T init() {
+    return T(1);
+  }
+};
+
+struct sca_two {
+  template <typename T>
+  static T init() {
+    return T(2);
+  }
+};
+
+inline auto get_scalars() {
+  return named_type_pack<sca_one, sca_two>::generate("one (1)", "two (2)");
+}
+
+template <typename DataT, typename NumElementsT, typename OpT,
+          typename SequenceT>
+class run_unary_sequence {
+  using helper = operators_helper<DataT, NumElementsT>;
+
+ public:
+  void operator()(const std::string &function_name) {
```

Comment:
You changed again coding style.

Reply:
I take it that this refers to the alignment of the reference symbol, I have fixed this.